build.gradle

```diff
@@ -752,6 +752,7 @@ project(':core') {
   dependencies {
     compile project(':clients')
     compile project(':raft')
+    compile group: 'net.ripe.ipresource', name: 'ipresource', version: '1.47'
```

Comment:
Would be possible to validate if it's better to import just the bits of code we need for this validation to work instead of adding a whole dependency?

Reply:
@jeqo Not sure, I'll check all the magic done by _contains_ method this weekend.

BTW I used that library because it comes from RIPE-NCC, an authority in this field, and it's being maintained. (https://github.com/RIPE-NCC/ipresource)

Reply:
@jeqo sorry for the delay.

I've been reviewing the code flow and I don't see it easily replaceable:

- ACL IP ranges are parsed and validated as slash notation(e.g. "10.0.0.0/16"), comma prefix notation (e.g. "10.0.0.0/16,/24,/25,/31") and ranges with dash (e.g. "10.0.0.0-10.1.2.3").
- Host IPs are parsed to be valid IPv4 and IPv6 adresses.
- For IPv6 cases, it's able to handle compressed IP adresses.
- Ranges and IPs validation check not only format but values (e.g. not out bounded values are allowed as 256.0.0.0)
- Compare probably is the easier part, ensure that ACL and host IP are the same type and check if host IP is inside the range.

I'd like to add that ipresource has a good test suite, it's covering a lot of cases https://github.com/RIPE-NCC/ipresource/tree/master/src/test/java/net/ripe/ipresource